.github/workflows/book.yml

```diff
@@ -0,0 +1,43 @@
+name: Book
+
+on:
+  push:
+    branches: [ main ]
+
+# Adapted from:
+# https://github.com/rust-lang/mdBook/wiki/Automated-Deployment%3A-GitHub-Actions#github-pages-deploy
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: Install mdbook
+      run: |
+        mkdir mdbook
+        curl -sSL https://github.com/rust-lang/mdBook/releases/download/v0.4.21/mdbook-v0.4.21-x86_64-unknown-linux-gnu.tar.gz | tar -xz --directory=./mdbook
```

Comment:
is there a nicer approach than using this hard-coded path?

Reply:
I don't think there's a URL to get the latest version; I borrowed this pattern from https://rust-lang.github.io/mdBook/continuous-integration.html#pre-compiled-binaries which mentions some tradeoffs: "Specifying a specific URL means you have to manually update your script to get a new version. This may be a benefit if you want to lock to a specific version. However, some users prefer to automatically get a newer version when they are published."

I think pinning is fine for us here since we have no particular need for the latest & greatest version.